python/cuml/tsa/arima.pyx

```diff
@@ -364,18 +369,30 @@ class ARIMA(Base):
         the CumlArrayDescriptors work
         """
 
-        # Compute the differenced series
         cdef uintptr_t d_y_ptr = self.d_y.ptr
         cdef uintptr_t d_y_diff_ptr = self._d_y_diff.ptr
         cdef handle_t* handle_ = <handle_t*><size_t>self.handle.getHandle()
-        batched_diff(handle_[0], <double*> d_y_diff_ptr, <double*> d_y_ptr,
-                     <int> self.batch_size, <int> self.n_obs, self.order)
-
-        # Create a version of the order for the differenced series
         cdef ARIMAOrder cpp_order_diff = self.order
-        cpp_order_diff.d = 0
-        cpp_order_diff.D = 0
-        self.order_diff = cpp_order_diff
+
+        # Detect missing observations
```

Comment:
Is there any significant performance penalty for always checking for missing observations? If so, it might be good to add a parameter to give the user the option of skipping it if they desire, if not then that’s probably not needed

Reply:
The cost is neglectible. I don't mind adding an option to skip this check but I would set the default to doing the check because by default we use pre-differencing which doesn't work with missing observations.

Reply:
Since the cost is negligible I think we can just leave it as is